src/main/java/org/folio/rest/impl/PurchaseOrderHelper.java

```diff
@@ -130,13 +131,24 @@
       );
   }
 
-  private CompositePoLine findCorrespondingCompositePoLine(CompositePoLine poLine, JsonArray poLinesFromStorage) {
-    return poLinesFromStorage.copy().stream()
+
+  public CompletableFuture<JsonObject> validateIfPOProtectedFieldsChanged(CompositePurchaseOrder compPO,
+      JsonObject compPOFromStorage) {
+    if (!compPOFromStorage.getString(WORKFLOW_STATUS)
+      .equals(CompositePurchaseOrder.WorkflowStatus.PENDING.value())) {
+      verifyProtectedFieldsChanged(POProtectedFields.getFieldNames(), compPOFromStorage, JsonObject.mapFrom(compPO));
+    }
+    return completedFuture(compPOFromStorage);
+  }
+
+  private JsonObject findCorrespondingCompositePoLine(CompositePoLine poLine, JsonArray poLinesFromStorage) {
+    return poLinesFromStorage.copy()
```

Comment:
The `copy` operation might not be needed. In case there is a reason why we need to work with copy of original object, I would advise to add as the last step (right before `.orElse`) the following: `.map(JsonObject::copy)`